### PR TITLE
Water uptake part 4.

### PR DESCRIPTION
**Pull request by pressel**
This is the fourth PR associated with water uptake. It provides a port of the `modal_aero_water_uptake_dryer` subroutine.

### CODE REVIEW

File: src/mam4xx/water_uptake.hpp [hunk 1]
```diff
@@ -301,6 +306,129 @@ void modal_aero_water_uptake_rh_clearair(const Real temperature,
...
+
+  const int ntot_amode = AeroConfig::num_modes();
+  int nspec_amode_temp[ntot_amode] = {7, 4, 7, 3};
+
+  for (int i = 0; i < ntot_amode; ++i) {
+    nspec_amode[i] = nspec_amode_temp[i];
+  }
+
+  Real specdens_amode_temp[maxd_aspectype] = {
+      0.1770000000E+04, 0.1797693135 + 309, 0.1797693135 + 309,
```
Comment (jeff-cohere) [inline src/mam4xx/water_uptake.hpp:323]: The second and third values here look fishy to me (and also those on line 329), but they're the same, so it must be intentional(?).

File: src/mam4xx/water_uptake.hpp [hunk 2]
```diff
@@ -58,7 +59,11 @@ class Water_Uptake {
 };
 
 namespace water_uptake {
-
+//  number of variables in state_q
+constexpr int nvars = 40;
+constexpr int maxd_aspectype = 14;
```
Comment (jeff-cohere) [inline src/mam4xx/water_uptake.hpp:64]: is this the number of distinct interstitial + cloudeborne species indices?

**Review by jeff-cohere — APPROVED** [review]
Looks good.

File: src/mam4xx/water_uptake.hpp (hunk 1 above)
Comment (pressel) [reply src/mam4xx/water_uptake.hpp:323]: Yeah, they looked fishy to me too but are values output directly by e3sm, see[ here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/c658ed322a9cc2abb9fe6dce85d45cebe872ac00/components/eam/src/chemistry/yaml/cam_ndrop/dropmixnuc_input_ts_1400.yaml).

@odiazib

File: src/mam4xx/water_uptake.hpp (hunk 2 above)
Comment (pressel) [reply src/mam4xx/water_uptake.hpp:64]: According to comments in the F90 code, it is the number of  vapor and aerosol number/mass mixing ratio variables in state%q [#/kg or kg/kg].

**Pull request merged**